### PR TITLE
feat(): add attribution_clients template to mobile_kpi_support_metrics sql generator

### PR DESCRIPTION
**Pull request by kik-kik**
# feat(): add attribution_clients template to mobile_kpi_support_metrics sql generator

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-4385)

### CODE REVIEW

File: sql_generators/mobile_kpi_support_metrics/__init__.py [hunk 1]
```diff
@@ -19,13 +19,15 @@
...
-    "engagement_clients.view.sql",
-    "engagement.query.sql",
-    "engagement.view.sql",
+    # "active_users.view.sql",
+    # "retention_clients.view.sql",
+    # "retention.query.sql",
+    # "retention.view.sql",
+    # "engagement_clients.view.sql",
+    # "engagement.query.sql",
+    # "engagement.view.sql",
```
Comment (bani) [inline sql_generators/mobile_kpi_support_metrics/__init__.py:28]: are you going to uncomment before merging?

File: sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.schema.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{{ header }}
```
Comment (bani) [inline sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.schema.yaml:1]: I don't think the schema is supposed to have a query?

File: sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.view.sql [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{{ header }}
+CREATE OR REPLACE VIEW
```
Comment (bani) [inline sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.view.sql:2]: Should we have submission_date on the view? In particular given it's the partition for the table

File: sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.query.sql [hunk 4]
```diff
@@ -0,0 +1,225 @@
+{{ header }}
+WITH new_profiles AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+  FROM `{{ project_id }}.{{ dataset }}.active_users`
```
Comment (bani) [inline sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.query.sql:7]: is there a reason why we are reading from active_users?
While the data is there, it doesn't seem we are leveraging anything specific to active_users in this query, and it's really hard to trace back the new profile logic from here.
Also, this is a table that has all clients active in the last 28 days, so it's unnecessarily large for this query

**Comment by bani** [comment]
I think this is from the previous PR, but there was a UNIONed view for attribution_clients and I noticed two things in it:
1. it was accounting for `is_suspicious_device_client` but I don't see that in the source tables you created now
2. Query looked a bit weird first with SELECT * EXCEPT and then placing the fields back
3. The fields changed to STRUCT but you are still querying with the old format

File: sql_generators/mobile_kpi_support_metrics/__init__.py (hunk 1 above)
Comment (kik-kik) [reply sql_generators/mobile_kpi_support_metrics/__init__.py:28]: yes, that's my bad. Thanks

File: sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.schema.yaml (hunk 2 above)
Comment (kik-kik) [reply sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.schema.yaml:1]: sorry, copy and paste mistake. Updated.

File: sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.view.sql (hunk 3 above)
Comment (kik-kik) [reply sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.view.sql:2]: I don't think it's strictly necessary here and partitioning would still be used correctly when getting the data from the underlying table if someone applied the filter on the view. For now I added it to reduce the confusion.

**Comment by kik-kik** [comment]
> I think this is from the previous PR, but there was a UNIONed view for attribution_clients and I noticed two things in it:
> 
>     1. it was accounting for `is_suspicious_device_client` but I don't see that in the source tables you created now
> 
>     2. Query looked a bit weird first with SELECT * EXCEPT and then placing the fields back
> 
>     3. The fields changed to STRUCT but you are still querying with the old format

So regarding 1. this indeed appears to be missing in the generated attribution table. I will take a look tomorrow and get this resolved.

2. This is necessary in the union as different products have different fields which means these fields are also present in their schemas in a different order. I need to do this EXCEPT and add them to the SELECT in order to ensure in the view all of the product specific selects get the fields in the same order. Alternative approach here would be to not hav ethe SELECT * EXCEPT and have each field explicitly listed. @bani would that be a preferred approach?
3. 
4. The attribution fields are a struct in the new attribution table, but the view flattens these out. Would you prefer if we already did that in the query?

File: sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.query.sql (hunk 4 above)
Comment (kik-kik) [reply sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.query.sql:7]: In this case, I think we could also change this to the baseline_clients_first_seen view to act as the source.

**Comment by bani** [comment]
I see. No problem with the current EXCEPT/SELECT and with the STRUCT flattened out.

File: sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.schema.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+- mode: NULLABLE
+  name: submission_date
+  type: DATE
+  description:
+
+- mode: NULLABLE
+  name: client_id
+  type: STRING
+  description:
+
```
Comment (bani) [inline sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.schema.yaml:11]: missing sample_id

File: sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.query.sql [hunk 6]
```diff
@@ -0,0 +1,240 @@
+{{ header }}
+WITH new_profiles AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    {% if 'is_suspicious_device_client' in product_attribution_group_names %}
```
Comment (bani) [inline sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.query.sql:7]: I see it on the table but not on the view

Comment (kik-kik) [reply sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.query.sql:7]: You are correct. I added it now.

File: sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.schema.yaml (hunk 5 above)
Comment (kik-kik) [reply sql_generators/mobile_kpi_support_metrics/templates/attribution_clients.schema.yaml:11]: added.

**Review by bani — APPROVED** [review]

**Pull request merged**